orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/HealthHelper.groovy

```diff
@@ -96,22 +98,19 @@ class HealthHelper {
     }
 
     // no health indicators is indicative of being down
-    boolean someAreDown = !healths || healths.any { it.state == 'Down' || it.state == 'OutOfService' || it.state == 'Starting' }
-    boolean noneAreUp = !healths.any { it.state == 'Up' }
+    boolean someAreDown = !healths || healths.any { HealthState.fromString(it.state) in [Down, OutOfService, Starting] }
+    boolean noneAreUp = !healths.any { HealthState.fromString(it.state) == Up }
 
     return someAreDown && noneAreUp
   }
 
   static boolean someAreUpAndNoneAreDownOrStarting(Map instance, Collection<String> interestingHealthProviderNames) {
     List<Map> healths = filterHealths(instance, interestingHealthProviderNames)
-    boolean someAreUp = healths.any { Map health -> health.state == 'Up' }
-    someAreUp = areSomeUpConsideringPlatformHealth(healths, interestingHealthProviderNames, someAreUp)
 
-    boolean noneAreDownOrStarting = !healths.any {
-      Map health -> health.state == 'Down' || health.state == 'Starting'
-    }
+    boolean someAreUp = healths.any { HealthState.fromString(it.state) == Up }
+    boolean noneAreDown = !healths.any { HealthState.fromString(it.state) in [Down, OutOfService, Starting] }
```

Comment:
worth making upness/downness an attribute of the enum?  (and I think Failed is maybe a down but I dunno..)

Reply:
I was thinking about that, it probably makes sense. `Failed` is referenced in the k8s, dcos and azure providers, it seems like it should be a down status indeed.